File: user-service/src/main/java/com/sonas/userservice/dao/Social.java
package com.sonas.userservice.dao;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.net.URL;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Social {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "social_id")
    private long socialId;

    private String linkType;

    private URL link;

    private long contactId;

    public Social(String linkType, URL link, long contactId) {
        this.linkType = linkType;
        this.link = link;
        this.contactId = contactId;
    }
}

File: sonasFrontEnd/sonas/src/app/models/social.ts
export class Social {

    public get linkType(): string {
        return this.linkType;
    }
    public set linkType(value: string) {
        this.linkType = value;
    }

    public get link(): URL {
        return this.link;
    }
    public set link(value: URL) {
        this.link = value;
    }

    public get contactId(): number {
        return this.contactId;
    }
    public set contactId(value: number) {
        this.contactId = value;
    }
}